src/mc-range-edge-generation-private.h

```diff
@@ -75,24 +77,89 @@ typedef struct {
 mc_edges_t *
 mc_getEdgesDouble (mc_getEdgesDouble_args_t args, mongocrypt_status_t *status);
 
+typedef struct {
+   mc_dec128 value;
+   size_t sparsity;
+   mc_optional_dec128_t min, max;
+   mc_optional_uint32_t precision;
+} mc_getEdgesDecimal128_args_t;
+
+mc_edges_t *
+mc_getEdgesDecimal128 (mc_getEdgesDecimal128_args_t args,
+                       mongocrypt_status_t *status);
+
+
 // count_leading_zeros_u64 returns the number of leading 0 bits of `in`.
-size_t
-mc_count_leading_zeros_u64 (uint64_t in);
+static inline size_t
+mc_count_leading_zeros_u64 (uint64_t in)
+{
+#ifdef __has_builtin
+#if __has_builtin(__builtin_clzl)
+// Pointer-cast to ensure we are speaking the right type
+#ifdef __APPLE__
+   unsigned long long *p = &in;
```

Comment:
The pointer conversion is to compile-time assert that `in` and `p` are an identical type (since uint64_t is an "implementation defined" `typedef`). If they aren't the same type, it will result in an incompatible-pointers warning (which we set to be an error).